toonz/sources/tnztools/brushtool.cpp

```diff
@@ -358,9 +359,14 @@ static void addStroke(TTool::Application *application, const TVectorImageP &vi,
   // Up this value the point can be considered a max curvature point.
 
   findMaxCurvPoints(stroke, angoloLim, curvMaxLim, corners);
-  TXshSimpleLevel *sl = application->getCurrentLevel()->getSimpleLevel();
+  TXshSimpleLevel *sl;
+  if (!sLevel) {
+    sl = application->getCurrentLevel()->getSimpleLevel();
+  } else {
+    sl = sLevel;
+  }
   TFrameId id = application->getCurrentTool()->getTool()->getCurrentFid();
-
+  if (id == -2 && fid != -2) id = fid;
```

Comment:
Confirmed this PR fixes the problem. Thank you !

How about using `TFrameId::NO_FRAME` instead of a magic number `-2` ?